packages/app/ui/price.tsx

```diff
@@ -0,0 +1,48 @@
+import { Pressable, Text, View } from "app/design-system";
+import Dollar from "app/ui/icons/dollar";
+import { tw } from "app/design-system/tailwind";
+import useEntryPrice from "app/hooks/useEntryPrice";
+import { openURL } from "expo-linking";
+import { stellarAssetLink } from "app/utils/stellar";
+
+type PriceProps = PriceOfEntryProps | PriceFrontProps;
+
+export default function Price(props: PriceProps) {
+  if ("code" in props && "issuer" in props) {
+    return <PriceOfEntry {...props} />;
+  }
+  return <PriceFront {...props} />;
```

Comment:
I don't like this approach. Can we make props for the price the same for all the cases?